controllers/chaincodebuild/predict.go

```diff
@@ -0,0 +1,136 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * 	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package chaincodebuild
+
+import (
+	"context"
+	"fmt"
+
+	current "github.com/IBM-Blockchain/fabric-operator/api/v1beta1"
+	"github.com/go-test/deep"
+	"github.com/pkg/errors"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+func (r *ReconcileChaincodeBuild) CreateFunc(e event.CreateEvent) bool {
+	chaincodeBuild := e.Object.(*current.ChaincodeBuild)
+	log.Info(fmt.Sprintf("Create event detected for ChaincodeBuild '%s'", chaincodeBuild.GetName()))
+
+	update := Update{}
+
+	if chaincodeBuild.HasType() {
+		log.Info(fmt.Sprintf("Operator restart detected, running update flow on existing ChaincodeBuild '%s'", chaincodeBuild.GetName()))
+
+		// Get the spec state of the resource before the operator went down, this
+		// will be used to compare to see if the spec of resources has changed
+		cm, err := r.GetSpecState(chaincodeBuild)
+		if err != nil {
+			log.Info(fmt.Sprintf("Failed getting saved ChaincodeBuild spec '%s', triggering create: %s", chaincodeBuild.GetName(), err.Error()))
+			return true
+		}
+
+		specBytes := cm.BinaryData["spec"]
+		existingchaincodeBuild := &current.ChaincodeBuild{}
+		err = yaml.Unmarshal(specBytes, &existingchaincodeBuild.Spec)
+		if err != nil {
+			log.Info(fmt.Sprintf("Unmarshal failed for saved ChaincodeBuild spec '%s', triggering create: %s", chaincodeBuild.GetName(), err.Error()))
+			return true
+		}
+
+		diff := deep.Equal(chaincodeBuild.Spec, existingchaincodeBuild.Spec)
+		if diff != nil {
+			log.Info(fmt.Sprintf("ChaincodeBuild '%s' spec was updated while operator was down", chaincodeBuild.GetName()))
+			log.Info(fmt.Sprintf("Difference detected: %v", diff))
+			update.specUpdated = true
+			update.pipelineSpecUpdated = true
+		}
+
+		log.Info(fmt.Sprintf("Create event triggering reconcile for updating ChaincodeBuild '%s'", chaincodeBuild.GetName()))
+		r.PushUpdate(chaincodeBuild.GetName(), update)
+		return true
+	}
+
+	update.specUpdated = true
+	update.pipelineSpecUpdated = true
+
+	r.PushUpdate(chaincodeBuild.GetName(), update)
+
+	return true
+}
+
+func (r *ReconcileChaincodeBuild) PipelineRunUpdateFunc(e event.UpdateEvent) bool {
+	// oldPipelineRun := e.ObjectOld.(*pipelinev1beta1.PipelineRun)
+	newPipelineRun := e.ObjectOld.(*pipelinev1beta1.PipelineRun)
```

Comment:
Should it be `newPipelineRun := e.ObjectNew.(*pipelinev1beta1.PipelineRun)`?

Reply:
yes...